odoo/tools/image.py

```diff
@@ -149,7 +149,10 @@ def image_quality(self, quality=0, output_format=''):
         if output_image.mode not in ["1", "L", "P", "RGB", "RGBA"] or (output_format == 'JPEG' and output_image.mode == 'RGBA'):
             output_image = output_image.convert("RGB")
 
-        return image_apply_opt(output_image, **opt)
+        output_bytes = image_apply_opt(output_image, **opt)
+        if len(output_bytes) >= len(self.source) and self.original_format == output_format and not self.operationsCount:
+            return self.source
+        return output_bytes
```

Comment:
I think a comment wouldn't hurt here :+1: 
Also I'm not sure why there is `and not self.operationsCount`, can you explain ?